cpp/src/arrow/compute/kernels/vector_selection.cc

```diff
@@ -2146,6 +2147,144 @@ class TakeMetaFunction : public MetaFunction {
   }
 };
 
+// ----------------------------------------------------------------------
+// DropNull Implementation
+
+Result<std::shared_ptr<arrow::Array>> GetNotNullIndices(
+    const std::shared_ptr<Array>& column, MemoryPool* memory_pool) {
+  std::shared_ptr<arrow::Array> indices;
+  arrow::NumericBuilder<arrow::Int32Type> builder(memory_pool);
+  builder.Reserve(column->length() - column->null_count());
```

Comment:
Every function which returns Status or Result<> will prompt a similar compiler error if not explicitly handled, since it could be ignoring (for example) an OOM Status https://github.com/apache/arrow/pull/10802/checks?check_run_id=3202879387#step:5:608
```suggestion
  RETURN_NOT_OK(builder.Reserve(column->length() - column->null_count()));
```